src/tgp.cpp

```diff
@@ -183,21 +183,36 @@ struct HeightMap
 static HeightMap _height_map = { {}, 0, 0, 0 };
 
 /** Conversion: int to Height */
-#define I2H(i) ((i) << HEIGHT_DECIMAL_BITS)
+static Height I2H(int i)
+{
+	return i << HEIGHT_DECIMAL_BITS;
+}
+
 /** Conversion: Height to int */
-#define H2I(i) ((i) >> HEIGHT_DECIMAL_BITS)
+static int H2I(Height i)
+{
+	return i >> HEIGHT_DECIMAL_BITS;
+}
 
 /** Conversion: Amplitude to int */
-#define A2I(i) ((i) >> AMPLITUDE_DECIMAL_BITS)
+static int A2I(Amplitude i)
+{
+	return i >> AMPLITUDE_DECIMAL_BITS;
+}
 
 /** Conversion: Amplitude to Height */
-#define A2H(a) ((a) >> (AMPLITUDE_DECIMAL_BITS - HEIGHT_DECIMAL_BITS))
+static Height A2H(Amplitude i)
+{
+	return I2H(A2I(i));
```

Comment:
You have
`(i >> AMPLITUDE_DECIMAL_BITS) << HEIGHT_DECIMAL_BITS`
it should be
`i >> (AMPLITUDE_DECIMAL_BITS - HEIGHT_DECIMAL_BITS)`